From 024091d65c03dcc96e2d5fc29cda193e8e84fef8 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Tue, 10 Dec 2024 11:17:41 +0100
Subject: [PATCH 1450/1695] plat-stm32mp1: CFG_STM32_IWDG requires
 CFG_WDT_SM_HANDLER

Driver stm32_iwdg only aims at exposing an OP-TEE watchdog
service hence declare CFG_STM32_IWDG dependency on CFG_WDT and
CFG_WDT_SM_HANDLER in stm32mp1 platform configuration file.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git e87388a0857f2a961625881a444be9584feda6a6 plat-stm32mp1: CFG_STM32_IWDG requires CFG_WDT_SM_HANDLER]
Change-Id: I19d75394fffe525e7df54b0f7e38821e07965ec0
Link: https://github.com/OP-TEE/optee_os/commit/d97509bff0d0396246944cfcf930abfc9471335
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/434833
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/conf.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/arch/arm/plat-stm32mp1/conf.mk b/core/arch/arm/plat-stm32mp1/conf.mk
index 9a885282a..23ccb355c 100644
--- a/core/arch/arm/plat-stm32mp1/conf.mk
+++ b/core/arch/arm/plat-stm32mp1/conf.mk
@@ -384,6 +384,7 @@ $(eval $(call cfg-depends-all,CFG_STM32_RSTCTRL,CFG_DRIVERS_RSTCTRL))
 CFG_WDT ?= $(CFG_STM32_IWDG)
 CFG_WDT_SM_HANDLER ?= $(CFG_WDT)
 CFG_WDT_SM_HANDLER_ID ?= 0xbc000000
+$(eval $(call cfg-depends-all,CFG_STM32_IWDG,CFG_WDT_SM_HANDLER CFG_WDT))
 
 # Platform specific configuration
 CFG_STM32MP_PANIC_ON_TZC_PERM_VIOLATION ?= y
-- 
2.47.2

